### PR TITLE
Attempt to get HPC target green

### LINKED ISSUE #12506: Test failing on falcon porous_flow/test:jacobian.eff_stress04
## Rationale
```
porous_flow/test:jacobian.eff_stress04: [0]PETSC ERROR: --------------------- Error Message --------------------------------------------------------------
porous_flow/test:jacobian.eff_stress04: [0]PETSC ERROR: Object is in wrong state
porous_flow/test:jacobian.eff_stress04: [0]PETSC ERROR: SNESTest aborts after Jacobian test: it is NORMAL behavior.
porous_flow/test:jacobian.eff_stress04: [0]PETSC ERROR: See http://www.mcs.anl.gov/petsc/documentation/faq.html for trouble shooting.
porous_flow/test:jacobian.eff_stress04: [0]PETSC ERROR: Petsc Release Version 3.8.3, Dec, 09, 2017 
porous_flow/test:jacobian.eff_stress04: [0]PETSC ERROR: /home/slauae/moose/modules/combined/combined-opt on a arch-linux2-c-opt named r5i5n14 by slauae Fri Nov 16 16:19:14 2018
porous_flow/test:jacobian.eff_stress04: [0]PETSC ERROR: Configure options --prefix=/apps/moose/gnu/petsc/petsc-3.8.3 --download-hypre=1 --with-ssl=0 --with-debugging=no --with-pic=1 --with-shared-libraries=0 --with-cc=mpicc --with-cxx=mpicxx --with-fc=mpif90 --download-fblaslapack=1 --download-metis=1 --download-parmetis=1 --download-superlu_dist=1 --download-mumps=1 --download-scalapack=1 -CC=mpicc -CXX=mpicxx -FC=mpif90 -F77=mpif77 -F90=mpif90 -CFLAGS="-fPIC -fopenmp" -CXXFLAGS="-fPIC -fopenmp" -FFLAGS="-fPIC -fopenmp" -FCFLAGS="-fPIC -fopenmp" -F90FLAGS="-fPIC -fopenmp" -F77FLAGS="-fPIC -fopenmp" PETSC_DIR=/apps/moose/stack_src/petsc-3.8.3
porous_flow/test:jacobian.eff_stress04: [0]PETSC ERROR: #1 SNESSolve_Test() line 173 in /apps/moose/stack_src/petsc-3.8.3/src/snes/impls/test/snestest.c
porous_flow/test:jacobian.eff_stress04: [0]PETSC ERROR: #2 SNESSolve() line 4109 in /apps/moose/stack_src/petsc-3.8.3/src/snes/interface/snes.c
porous_flow/test:jacobian.eff_stress04: [unset]: aborting job:
porous_flow/test:jacobian.eff_stress04: application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
porous_flow/test:jacobian.eff_stress04: 
porous_flow/test:jacobian.eff_stress04: ################################################################################
porous_flow/test:jacobian.eff_stress04: Tester failed, reason: INCORRECT JACOBIAN
```

## Description
The error can be reproduced on falcon with the following modules

```
Currently Loaded Modules:
  1) use.moose   2) GCC/4.9.2   3) cppunit/1.12.1-GCC-4.9.2   4) tbb/4.3.0.090   5) MVAPICH2/2.0.1-GCC-4.9.2   6) moose-dev-gcc_petsc-3.8.3
``` 

## Impact
Unknown

### PR DESCRIPTION
**Pull request by aeslaughter**
The HPC falcon testing needs to be passing so we can make failures fatal on CIVET to prevent 
regressions entering.

Closes #12506

### CODE REVIEW

**Comment by aeslaughter** [comment]
@bwspenc @WilkAndy I did some work to get our HPC target working, which it is. However, I had to touch some test specifications, so now requirements are needed. I would write them, but I don't know much about these tests so they wouldn't be very good. Can I get some help please?

**Comment by WilkAndy** [comment]
Hopefully you have a PR to your branch, @aeslaughter , which fixes the PorousFlow tests

File: modules/combined/test/tests/contact_verification/patch_tests/single_pnt_2d/tests [hunk 1]
```diff
@@ -2,6 +2,7 @@
   issues = '#716'
   design = 'syntax/Contact/index.md source/MechanicalContactConstraint.md'
   [./glued_kin]
+    skip = '#12508'
```
Comment (bwspenc) [inline modules/combined/test/tests/contact_verification/patch_tests/single_pnt_2d/tests]: You shouldn't need to skip this one.

File: modules/heat_conduction/test/tests/meshed_gap_thermal_contact/tests [hunk 2]
```diff
@@ -14,6 +14,7 @@
   [../]
 
   [./annulus]
+    skip = '#12499'
```
Comment (bwspenc) [inline modules/heat_conduction/test/tests/meshed_gap_thermal_contact/tests]: This one is fixed in my PR.

File: modules/combined/test/tests/eigenstrain/tests [hunk 3]
```diff
@@ -39,8 +39,6 @@
     input = 'inclusion.i'
     exodiff = 'inclusion_out.e'
     fparser_jit = True
-    issues = '#7452'
-    design = 'ComputeVariableEigenstrain.md InclusionProperties.md'
-    requirement = "The InclusionProperties material, in conjunction with the ComputeVariableEigenstrain material, shall correctly compute the in-plane dilitational eigenstrains resulting from an ellipsoidal inclusion."
+    abs_zero = 1e-8
```
Comment (bwspenc) [inline modules/combined/test/tests/eigenstrain/tests]: Fix this

File: modules/tensor_mechanics/test/tests/capped_weak_plane/tests [hunk 4]
```diff
@@ -110,6 +110,7 @@
     type = Exodiff
     input = 'pull_and_shear_1step.i'
     exodiff = 'pull_and_shear_1step.e'
+    abs_zero = 1e-7
```
Comment (aeslaughter) [inline modules/tensor_mechanics/test/tests/capped_weak_plane/tests]: @bwspenc Can you write some requirements for these tests or do you want me to try to make something up?

Comment (aeslaughter) [reply modules/tensor_mechanics/test/tests/capped_weak_plane/tests]: FYI, I am working on falcon now to see if I can get this solve to be a bit more robust.

Comment (aeslaughter) [reply modules/tensor_mechanics/test/tests/capped_weak_plane/tests]: OK, I am done. I just made a comparison file. There were alot of variables that are zero being compared.

Comment (bwspenc) [reply modules/tensor_mechanics/test/tests/capped_weak_plane/tests]: I can take a stab at it.  I really don't know anything about this model, and there are a ton of tests in this file. Is this the last one?

Comment (aeslaughter) [reply modules/tensor_mechanics/test/tests/capped_weak_plane/tests]: This should be the last one, this is actually a new one that showed up after changing the default environment on the build machine. Thank you.

Comment (bwspenc) [reply modules/tensor_mechanics/test/tests/capped_weak_plane/tests]: @aeslaughter I added those SQA requirements in PR #12569, and @WilkAndy says they look OK. You can either merge that and wait for it to go in to devel, or cherry-pick that commit into your branch for this PR, whichever is easier for you. Hopefully we're almost there!

File: modules/combined/test/tests/contact_verification/patch_tests/single_pnt_2d/tests [hunk 5]
```diff
@@ -72,6 +72,7 @@
     requirement = "The Contact system shall enforce frictionless, penalty contact
     for 2D plane strain single point contact model using the contact line
     search solver options."
+    compiler = 'GCC CLANG'
```
Comment (bwspenc) [inline modules/combined/test/tests/contact_verification/patch_tests/single_pnt_2d/tests]: Does this one fail on Intel or something now?

**Review by bwspenc — APPROVED** [review]

**Pull request merged**